### ISSUE TITLE
Wrong condition make Database schema migration always triggered

### ISSUE THREAD

**Issue by lamm** [ansible/tower-operator#1949]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

Operator always try to migrate schema even if there are no pending migration.

### AWX Operator version

2.15.0

### AWX version

24.2.0

### Kubernetes platform

openshift

### Kubernetes/Platform version

4.14.34

### Modifications

yes

### Steps to reproduce

Any modification that make the operator to [include migrate_schema.yml](https://github.com/ansible/awx-operator/blob/f0a518baf6ae830b6d1ac321bc1661d7566615f8/roles/installer/tasks/install.yml#L98)

### Expected results

The operator should not execute the [block](https://github.com/ansible/awx-operator/blob/f0a518baf6ae830b6d1ac321bc1661d7566615f8/roles/installer/tasks/migrate_schema.yml#L14) which execute the schema migration process if no migration is pending.

### Actual results

The operator execute the schema migration even if there are no pending migrations.

### Additional information

When looking at the command executed to count the number of pending migrations, i noticed that it does not filter as it should.

```
sh-5.1$ bash -c "awx-manage showmigrations | grep -v '[X]' | grep '[ ]'"
 (no migrations)
 (no migrations)
```
```
sh-5.1$ bash -c "awx-manage showmigrations | grep -v '[X]'"
auth
conf
contenttypes
dab_jwt_consumer
 (no migrations)
dab_resource_registry
dab_rest_filters
 (no migrations)
main
oauth2_provider
sessions
sites
social_django
sso
```

Lines containing ___(no migrations)___ are not filtered. That's why the schema migration process is executed.

**Comment by oziel88** [comment]
Yeah seeing the same issue after upgrading from operator 2.0.0 (awx 22.0.0) to 2.19.1 (awx 24.6.1)

The upgrade worked good after I ran command:

```
awx-manage migrate --noinput
```

But now I keep getting this pod below in error state 

```
NAME                                               READY   STATUS    RESTARTS   AGE
awx-migration-24.6.1-27hbp                         1/2     Error     0          36m
```

For now will just update migrate_schema.yml to filter out (no migrations)

```
bash-5.1$ bash -c "awx-manage showmigrations | grep -v '[X]' | grep '[ ]' | wc -l"
2
bash-5.1$ bash -c "awx-manage showmigrations | grep -v '[X]' | grep '[ ]'"
 (no migrations)
 (no migrations)
bash-5.1$ bash -c "awx-manage showmigrations | egrep -v '[X]|(no migrations)' | grep '[ ]' | wc -l"
0
bash-5.1$
```
